AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py

```diff
@@ -58,101 +63,74 @@ class ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
-                        self.validate_entity_state(deactivated_dict, Tests.deactivated_exists) and
-                        self.validate_entity_state(controller_dict, Tests.controller_exists))
-
-        Report.critical_result(Tests.start_states_correct, entities_valid)
-
-
-    @pyside_utils.wrap_async
-    async def run_test(self):
-
-        # 1) Create temp level
-        general.idle_enable(True)
-        hydra.open_base_level()
-        helper.wait_for_condition(lambda: general.get_current_level_name() == BASE_LEVEL_NAME, WAIT_TIME_3)
-        general.close_pane("Error Report")
-
-        # 2) create all the entities we need for the test
-        self.setup_level_entities()
-
-        # 3) Validate the entities were created and configured
-        self.validate_test_entities()
-
-        # 4) Start the Tracer
-        with Tracer() as section_tracer:
-
-            # 5) Enter Game Mode
-            helper.enter_game_mode(Tests.game_mode_entered)
-
-            # 6 Wait one second for graph timers then exit game mode
-            general.idle_wait(WAIT_ONE_SECOND)
-            helper.exit_game_mode(Tests.game_mode_exited)
-
-            # 7) Validate Print message
-            found_expected_lines = scripting_tools.located_expected_tracer_lines(self, section_tracer, EXPECTED_LINES)
-            helper.wait_for_condition(lambda: found_expected_lines is not None, WAIT_TIME_3)
-
-        Report.result(Tests.lines_found, found_expected_lines)
-
-
-
-test = ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage()
-test.run_test()
+    import os
+    import azlmbr.paths as paths
+    import scripting_utils.scripting_tools as scripting_tools
+    from editor_python_test_tools.utils import TestHelper as TestHelper
+    from editor_python_test_tools.utils import Report, Tracer
+    import azlmbr.legacy.general as general
+    from scripting_utils.scripting_constants import (WAIT_TIME_3, WAIT_TIME_1)
+
+    controller_dict = {
+        "name": "Controller",
+        "status": "active",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "controller.scriptcanvas"),
+    }
+    activated_dict = {
+        "name": "ActivationTest",
+        "status": "inactive",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "activator.scriptcanvas"),
+    }
+    deactivated_dict = {
+        "name": "DeactivationTest",
+        "status": "active",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "deactivator.scriptcanvas"),
+    }
+
+    EXPECTED_LINES = ["Activator Script: Activated", "Deactivator Script: Deactivated"]
+
+    # Preconditions
+    general.idle_enable(True)
+
+    # 1) Create temp level
+    TestHelper.open_level("", "Base")
+
+    # 2) create all the entities we need for the test
+    setup_level_entities(activated_dict)
+    setup_level_entities(deactivated_dict)
+    setup_level_entities(controller_dict)
```

Comment:
Extract this into a function that takes a list of entities to setup and pass it all three objects as a list.

`setup_all_level_entities([activated, deactivated, controller])`

Reply:
Better solution, no new function needed. 

`setup_level_entities(for enetiy in [activated_dict, deactivated_dict, controller_dict])`

Reply:
did something similar to this. I'm not huge fan of a for loop as an argument but the loop is in the function and the input parameter is the list now.